include/libp2p/security/noise/crypto/message_patterns.hpp

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_INCLUDE_LIBP2P_SECURITY_NOISE_CRYPTO_MESSAGE_PATTERNS_HPP
+#define LIBP2P_INCLUDE_LIBP2P_SECURITY_NOISE_CRYPTO_MESSAGE_PATTERNS_HPP
+
+#include <string>
+#include <vector>
+
+namespace libp2p::security::noise {
+
+  enum class MessagePattern : int {  // int required here
```

Comment:
Would be great to see short comments explaining the patterns or a link to the docs.

Reply:
Nothing to put here. Just an implementation details which are not going to be used outside this particular noise protocol implementation. Transferred from go impl.